bestie/server/testresult.go

```diff
@@ -0,0 +1,220 @@
+	m := bazelBuildEvent.GetTestResult()
+	if m == nil {
+		return fmt.Errorf("Error extracting TestResult data from event message")
+	}
+
+	fmt.Printf("TestResult for %s: %s\n", stream.testName, m.GetStatus())
+	fmt.Printf("  run: %s\n", stream.run)
+	fmt.Printf("  buildId: %s\n", stream.buildId)
+	fmt.Printf("  invocationId: %s\n", stream.invocationId)
+	fmt.Printf("  invocationSha: %s\n\n", stream.invocationSha)
+
+	outputFiles := m.GetTestActionOutput()
+	var ofname, ofuri string
+	found := false
+	for _, of := range outputFiles {
+		ofname = of.GetName()
+		ofuri = of.GetUri()
+		if strings.HasSuffix(ofname, "outputs.zip") {
+			found = true
+			break
+		}
+	}
+	if !found {
+		return nil
+	}
+
+	// Strip off any file:// prefix from the URI to access the local file system path.
+	filePrefix := "file://"
+	if strings.HasPrefix(ofuri, filePrefix) {
+		ofuri = ofuri[len(filePrefix):]
+	}
+
+	// Process test metrics output file(s).
+	// Each output file contains a single (potentially large) protobuf message.
+	if err := extractZippedFiles(stream, ofuri); err != nil {
+		return fmt.Errorf("Error processing %s file: %s", ofname, err)
+	}
+
+	return nil
+}
+
+// Look for and return a []byte slice with its contents.
+func extractZippedFiles(stream bazelStream, zipFile string) error {
+	metricsFileRE := regexp.MustCompile(`(^test_metrics.pb$|^test_metrics[_-]+[[:word:]-]*[[:alnum:]]\.pb$)`)
+	summaryFileRE := regexp.MustCompile(`(^test_summary.pb$|^test_summary[_-]+[[:word:]-]*[[:alnum:]]\.pb$)`)
+	allRE := []*regexp.Regexp{metricsFileRE, summaryFileRE}
+
+	reader, err := zip.OpenReader(zipFile)
+	if err != nil {
+		return err
+	}
+	defer reader.Close()
+
+	for _, file := range reader.File {
+		if file.FileInfo().IsDir() {
+			continue
+		}
+
+		//  Look for a matching file name using regular expression patterns.
+		match := ""
+		for _, re := range allRE {
+			match = re.FindString(filepath.Base(file.Name))
+			if len(match) > 0 {
+				break
+			}
+		}
+		if len(match) == 0 {
+			continue
+		}
+		fmt.Printf("Found output file to process: %s\n", file.Name)
+
+		f, err := file.Open()
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+
+		// Read entire file contents into a byte slice.
+		data, err := ioutil.ReadAll(f)
+		if err != nil {
+			return err
+		}
+
+		// Extract all metrics from the file contents.
+		pResult, err := getTestMetricsFromFileData(&data)
+		if err != nil {
+			fmt.Printf("Error processing output file %s: %s\n", file.Name, err)
+			continue
+		}
+		if pResult != nil {
+			// Display the metric data on the console.
+			displayMetrics(pResult, 2)
+			// TODO: Upload test metrics to BigQuery database table.
+		}
+	}
+	return nil
+}
+
+// Extract the test metrics information from the protobuf message data read
+// from an output file. This is used for processing test_metrics*.pb and
+// test_summary*.pb output files.
+func getTestMetricsFromFileData(pbmsg *[]byte) (*metricTestResult, error) {
```

Comment:
nit: slices don't need to be passed by pointer if they're only read (only if you want to append to it - but in that case it might be better to return a slice rather than have effectively a slice outparam)

Reply:
OK, will fix in this PR.